client/src/Components/OfferDetails/index.jsx

```diff
@@ -1,6 +1,148 @@
+    this.setState({ applications: applicationsData });
+    // take logged in userInfo from local storage and store it in state
+    // const userInfo = localStorage.getItem('userInfo');
+    // this.setState(userInfo);
+  }
+
+  offerColor = status => {
+    switch (status) {
+      case 'completed':
+      case 'accepted':
+        return 'green';
+      case 'pending':
+        return '#1BA7E2';
+      case 'in_progress':
+        return 'orange';
+      case 'finished':
+      case 'refused':
+        return 'red';
+      default:
+        return null;
+    }
+  };
+
+  handleEndContract = () => {
+    // handle end contract login here....
+  };
+
+  render() {
+    const { offer } = this.state;
+    const { applications } = this.state;
+    const offerColor = this.offerColor(offer.status);
+    const { id: memberId } = this.state.userInfo;
+    const memberApplication = applications.filter(
+      application => application.member_id === memberId
+    );
+
+    return (
+      <Container className="page__container">
+        <Row className="offer-details__header">
+          <Col className="offer-details__header-col">
+            <span className="offer-details__position">{offer.position}</span>
+            <p className="offer-details__title">{offer.title}</p>
+          </Col>
+
+          <Col className="offer-details__status" md="auto">
+            <span style={{ color: offerColor }}>
+              {offer.status
+                ? offer.status.replace(/_/g, ' ').toUpperCase()
+                : null}
+            </span>
+            <Button
+              className="offet-details__end-button"
+              variant="danger"
+              onClick={this.handleEndContract}
+            >
+              End Contract
+            </Button>
+          </Col>
+        </Row>
+        <Row className="offer-details__row">
+          <Col xs lg="9" className="offer-details__description">
+            <Row>
+              <p>{offer.description}</p>
+            </Row>
+          </Col>
+          <Col xs lg="2">
+            <div>
+              <SideCard title="skills" items={offer.skills} />
+              <SideCard title="offer type" items={offer.offer_type} />
+            </div>
+          </Col>
+        </Row>
+        {memberId === offer.member_id ? (
+          // {/* applications */}
+          <div>
+            <Row className="offer-details__Applications-title">
+              Applications
+            </Row>
+            <Col xs lg="9" style={{ padding: '0px' }}>
+              {applications &&
+                applications.map(item => (
+                  <div key={item.member_id}>
+                    <ApplicationCard
+                      application={item}
+                      offerColor={this.offerColor}
+                    />
+                  </div>
+                ))}
+            </Col>
+          </div>
+        ) : (
+          // if logged in member is not offer owner
+          (!offer.status && (
```

Comment:
I think we agreed that the offer status default value is 'active'  so this won't work
I suggest you check if the logged in member have applied before hence `!memberApplication[0]`

Reply:
ok